src/providers/theme/theme.ts

```diff
@@ -0,0 +1,123 @@
+import { Injectable } from '@angular/core';
+import { StatusBar } from '@ionic-native/status-bar';
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
+
+// providers
+import { Logger } from '../logger/logger';
+import { PersistenceProvider } from '../persistence/persistence';
+import { PlatformProvider } from '../platform/platform';
+
+@Injectable()
+export class ThemeProvider {
+  private currentAppTheme: BehaviorSubject<string>;
+
+  public themes = {
+    'light-theme': {
+      backgroundColor: '#ffffff',
+      fixedScrollBgColor: '#f8f8f9',
+      walletDetailsBackgroundStart: '#ffffff',
+      walletDetailsBackgroundEnd: '#ffffff'
+    },
+    'dark-theme': {
+      backgroundColor: '#121212',
+      fixedScrollBgColor: '#121212',
+      walletDetailsBackgroundStart: '#121212',
+      walletDetailsBackgroundEnd: '#101010'
+    }
+  };
+
+  constructor(
+    private logger: Logger,
+    private persistenceProvider: PersistenceProvider,
+    private statusBar: StatusBar,
+    private platformProvider: PlatformProvider
+  ) {
+    this.logger.debug('ThemeProvider initialized');
+  }
+
+  public init() {
```

Comment:
I think it's not necessary since we the `persistentProvider` is loaded before of everything . Maybe it's a good practice to set the variable in the constructor.

Reply:
We can’t move the init function to the constructor, since the themeProvider is loaded at the beginning of the app, being in the app component. Moving it to the constructor doesn’t give time to fully prepare persistanceProvider, so we get a storage related error.

Reply:
Got it. Thanks for explanation.